lxd/storage/backend_lxd.go

```diff
@@ -205,26 +223,25 @@ func (b *lxdBackend) CreateInstanceFromImage(inst Instance, fingerprint string,
 		b.DeleteInstance(inst, op)
 	}()
 
-	filler := func(instanceMountPath string) error {
-		var tracker *ioprogress.ProgressTracker
-		if op != nil { // Not passed when being done as part of pre-migration setup.
-			metadata := make(map[string]interface{})
-			tracker = &ioprogress.ProgressTracker{
-				Handler: func(percent, speed int64) {
-					shared.SetProgressMetadata(metadata, "create_instance_from_image_unpack", "Unpack", percent, 0, speed)
-					op.UpdateMetadata(metadata)
-				}}
+	vol := b.newVolume(volType, drivers.ContentTypeFS, project.Prefix(inst.Project(), inst.Name()), nil)
+	if !b.driver.Info().OptimizedImages {
+		err = b.driver.CreateVolume(vol, b.imageFiller(fingerprint, op), op)
+		if err != nil {
+			return err
+		}
+	} else {
+		if !b.driver.HasVolume(drivers.VolumeTypeImage, fingerprint) {
+			err := b.CreateImage(fingerprint, op)
```

Comment:
@stgraber thanks for doing this. As b.CreateImage checks for HasVolume inside it, we could remove the HasVolume check above and simplify this part somewhat.

Reply:
Oh, indeed, we could just always call it. Can you put that in your next branch?

Reply:
@stgraber will do